apps/site/src/lib/api/blocks/db.ts

```diff
@@ -1,15 +1,44 @@
 import { ExpandedBlockMetadata } from "../../blocks";
 import { connectToDatabase } from "../mongodb";
-import { blocksDbCollectionName } from "./shared";
+import { blockDownloadsCollectionName, blocksDbCollectionName } from "./shared";
 
-const queryOptions = { projection: { _id: 0 } };
+const defaultProjection = { _id: 0 };
+
+const weeklyDownloadCountAggregationStage = [
+  {
+    $lookup: {
+      from: blockDownloadsCollectionName,
+      let: { id: "$_id" },
+      pipeline: [
+        {
+          $match: {
+            $expr: { $eq: ["$$id", "$blockId"] },
+            downloadedAt: {
+              $gte: new Date(
+                new Date().valueOf() - 7 * 60 * 60 * 24 * 1000,
+              ).toISOString(),
+            },
+          },
+        },
+      ],
+      as: "weeklyDownloads",
+    },
+  },
+  { $addFields: { downloads: { weekly: { $size: "$weeklyDownloads" } } } },
+  { $project: { weeklyDownloads: 0, ...defaultProjection } },
+];
```

Comment:
Does there exist a type in Mongo package which we can use here with `satisfies`?